consensus/spos/bls/subroundEndRound.go

```diff
@@ -376,15 +377,66 @@ func (sr *subroundEndRound) broadcastMiniBlocksAndTransactions() error {
 			return err
 		}
 
-		return sr.BroadcastMessenger().SetDataForDelayBroadcast(headerHash, miniBlocks, transactions)
+		metaMiniBlocks, metaTransactions := sr.extractMetaMiniBlocksAndTransactions(miniBlocks, transactions)
+
+		err = sr.BroadcastMessenger().SetDataForDelayBroadcast(headerHash, miniBlocks, transactions)
+		if err != nil {
+			return err
+		}
+
+		return sr.broadcast(metaMiniBlocks, metaTransactions)
 	}
 
-	err = sr.BroadcastMessenger().BroadcastMiniBlocks(miniBlocks)
-	if err != nil {
-		return err
+	return sr.broadcast(miniBlocks, transactions)
+}
+
+func (sr *subroundEndRound) extractMetaMiniBlocksAndTransactions(
```

Comment:
this became a super special think. might rethink sometime a refactor of consensus, to be more general.

Reply:
Maybe add a unit test :D

Reply:
Actually the caller method already do things which depend of shardCoordinator.SelfId(), as we wanted to keep only one implementation for all shards

Reply:
Could have become more general by providing the shard ID. For now it will suffice.